src/common/components/repositories-home/index.js

```diff
@@ -92,18 +96,18 @@ class RepositoriesHome extends Component {
   }
 
   render() {
-    // show spinner if snaps data was not yet fetched (snaps list is empty)
-    // (to avoid spinner during polling we are not checking `success` or `isFetching`
+    // show spinner if user has just signed in snaps data was not yet fetched
     //
-    // when snaps are loaded and user doesn't have any, they will be redirected
-    // to select repositories (so spinner won't be showing endlessly)
-    return !this.props.hasSnaps
+    // when snaps are loaded and user (who just signed in) doesn't have any,
+    // they will be redirected to select repositories
+    return (this.props.hasJustSignedIn && this.props.snaps.isFetching)
```

Comment:
Hm, what about the case where a user refreshes their browser on `/user/:user` rather than going through the landing page?  `hasSnaps` will be false until snaps load, but I don't think `hasJustSignedIn` will be true, which means we'll show them an empty repository list rather than a spinner for a while.